### PR TITLE
Support kotlin multiplatform project layout in PackageNaming rule

### LINKED ISSUE #397: PACKAGE_NAME_INCORRECT_PATH KMP support
I have a kotlin multiplatform module, and it has the following structure:
```
module
	build.gradle
	src
		androidMain
			com.org.module
				*.kt files
		iosMain
			com.org.module
				*.kt files
		commonMain
			com.org.module
				*.kt files
```

In this case PACKAGE_NAME_INCORRECT_PATH work is incorrect:
- package com.org.module
+ package com.org.commonMain.kotlin.com.org.module

### PR DESCRIPTION
**Pull request by petertrr**
### What's done:
* Added logic
* Added tests

This pull request closes #397

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/PackageNaming.kt [hunk 1]
```diff
@@ -105,10 +98,12 @@ class PackageNaming(private val configRules: List<RulesConfig>) : Rule("package-
         } else {
             // creating a real package name:
             // 1) getting a path after the base project directory (after "src" directory)
-            // 2) removing src/main/kotlin/java/e.t.c dirs and removing file name
+            // 2) removing src/main/kotlin/java/e.t.c dirs (additional check for multiplatform project layout) and removing file name
             // 3) adding company's domain name at the beginning
             val fileSubDir = filePathParts.subList(filePathParts.lastIndexOf(PACKAGE_PATH_ANCHOR), filePathParts.size - 1)
-                    .dropWhile { LANGUAGE_DIR_NAMES.contains(it) }
+                    .dropWhile {
+                        LANGUAGE_DIR_NAMES.contains(it) || it.endsWith("Main") || it.endsWith("Test")
```
Comment (akuleshov7) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/PackageNaming.kt:105]: let's set it explicitly, unfortunately...

**Review by akuleshov7 — APPROVED** [review]
lgtm

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter1/PackageNamingWarnTest.kt [hunk 2]
```diff
@@ -182,4 +182,33 @@ class PackageNamingWarnTest : LintTestBase(::PackageNaming) {
                 )
         )
     }
+
+    @Test
```
Comment (akuleshov7) [inline diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter1/PackageNamingWarnTest.kt:186]: add negative test for "myProjectMain "

Comment (petertrr) [reply diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter1/PackageNamingWarnTest.kt:186]: Added

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/PackageNaming.kt (hunk 1 above)
Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/PackageNaming.kt]: I've added a list of currently supported targets for KMP.

**Pull request merged**